homeassistant/components/evohome/__init__.py

```diff
@@ -160,136 +232,104 @@ def _handle_exception(err) -> bool:
-            if at_expires_str:
-                at_expires_dt = dt_util.parse_datetime(at_expires_str)
-            else:
-                at_expires_dt = None
-
-            return (refresh_token, access_token, at_expires_dt)
-
-        return (None, None, None)  # account switched: so tokens wont be valid
+        self.hass.add_job(self._save_auth_tokens())
 
     async def _save_auth_tokens(self, *args) -> None:
         # evohomeasync2 uses naive/local datetimes
         access_token_expires = _local_dt_to_aware(self.client.access_token_expires)
 
-        self._app_storage[CONF_USERNAME] = self.params[CONF_USERNAME]
-        self._app_storage[CONF_REFRESH_TOKEN] = self.client.refresh_token
-        self._app_storage[CONF_ACCESS_TOKEN] = self.client.access_token
-        self._app_storage[CONF_ACCESS_TOKEN_EXPIRES] = access_token_expires.isoformat()
+        app_storage = {CONF_USERNAME: self.client.username}
+        app_storage[REFRESH_TOKEN] = self.client.refresh_token
+        app_storage[ACCESS_TOKEN] = self.client.access_token
+        app_storage[ACCESS_TOKEN_EXPIRES] = access_token_expires.isoformat()
 
-        store = self.hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
-        await store.async_save(self._app_storage)
+        if self.client_v1 and self.client_v1.user_data:
+            app_storage[USER_DATA] = {
+                "userInfo": {"userID": self.client_v1.user_data["userInfo"]["userID"]},
+                "sessionId": self.client_v1.user_data["sessionId"],
+            }
+        else:
+            app_storage[USER_DATA] = None
+
+        await self._store.async_save(app_storage)
 
         self.hass.helpers.event.async_track_point_in_utc_time(
             self._save_auth_tokens,
             access_token_expires + self.params[CONF_SCAN_INTERVAL],
         )
 
-    async def update(self, *args, **kwargs) -> None:
-        """Get the latest state data of an entire evohome Location.
+    async def _update_v1(self, *args, **kwargs) -> None:
+        """Get the latest (high-precision) temperatures of the default Location."""
+        loc_idx = self.params[CONF_LOCATION_IDX]
 
-        This includes state data for a Controller and all its child devices, such as the
-        operating mode of the Controller and the current temp of its children (e.g.
-        Zones, DHW controller).
-        """
+        self.temps = None
+
+        try:
+            temps = list(await self.client_v1.temperatures(force_refresh=True))
+        except aiohttp.ClientError as err:
+            _handle_exception(err)
+        else:
+            if (
+                str(self.client_v1.location_id)
+                != self.client.locations[loc_idx].locationId
+            ):
+                _LOGGER.warning(
+                    "The v2 API's configured location doesn't match "
+                    "the v1 API's default location (there is more than one location). "
+                    "The '%s' feature will be disabled.",
```

Comment:
We don't end logging messages with period.

Reply:
There's two sentences!